src/ExpEvaluator.h

```diff
@@ -0,0 +1,88 @@
+/**
+ * @file ExpEvaluator.h
+ * @brief The ExpEvaluator class.
+ * @date April 9, 2015.
+ * @author William Boyd, MIT, Course 22 (wboyd@mit.edu)
+ */
+
+#ifndef EXPEVALUATOR_H_
+#define EXPEVALUATOR_H_
+
+#ifdef __cplusplus
+#define _USE_MATH_DEFINES
+#include <math.h>
+#include "log.h"
+#include "PolarQuad.h"
+#endif
+
+
+/**
+ * @class ExpEvaluator ExpEvaluator.h "src/ExpEvaluator.h"
+ * @brief This is a class for evaluating exponentials.
+ * @details The ExpEvaluator includes different algorithms to evaluate
+ *          exponentials with varying degrees of accuracy and speed. This
+ *          is a helper class for the Solver and its subclasses and it not
+ *          intended to be initialized as a standalone object.
+ */
+class ExpEvaluator {
+
+private:
+
+  /** A boolean indicating whether or not to use linear interpolation */
+  bool _interpolate;
+
+  /** The inverse spacing for the exponential linear interpolation table */
+  FP_PRECISION _inverse_exp_table_spacing;
+
+  /** The number of entries in the exponential linear interpolation table */
+  int _table_size;
+
+  /** The exponential linear interpolation table */
+  FP_PRECISION* _exp_table;
+
+  /** The PolarQuad object of interest */
+  PolarQuad* _polar_quad;
+
+  /** Twice the number of polar angles */
+  int _two_times_num_polar;
+  
+public:
+
+  ExpEvaluator();
+  virtual ~ExpEvaluator();
+
+  void setPolarQuadrature(PolarQuad* polar_quad);
+  void useInterpolation();
+  void useIntrinsic();
+
+  bool isUsingInterpolation();
+  FP_PRECISION getTableSpacing();
+  int getTableSize();
+  FP_PRECISION* getExpTable();
+
+  void initialize(double max_tau, double tolerance);
```

Comment:
Yes, I don't think there is a tight relationship between the tolerance on the source convergence and the individual exponential calculations. As a result, I think that this current implementation (greatly) exaggerates the size of the table that is needed, reducing cache performance.